website/views.py

```diff
@@ -3591,6 +3590,30 @@ def like_issue2(request, issue_pk):
     context["likes"] = total_votes
     return render(request, "includes/_likes2.html", context)
 
+@login_required(login_url="/accounts/login")
+def dislike_issue2(request, issue_pk):
+    context = {}
+    issue_pk = int(issue_pk)
+    issue = Issue.objects.get(pk=issue_pk)
+    userprof = UserProfile.objects.get(user=request.user)
+    if userprof in UserProfile.objects.filter(issue_downvoted=issue):
+        userprof.issue_downvoted.remove(issue)
+    else:
+        userprof.issue_downvoted.add(issue)
+    userprof.save()
```

Comment:
Do you need `save()` here?

Reply:
I don't think you need it. As long as you update the object's m2m relations you don't need to do explicit `save()` for the object itself.

https://docs.djangoproject.com/en/5.0/topics/db/examples/many_to_many/#many-to-many-relationships

Reply:
Should i remove save() from both the views, i.e. like_issue2() and dislike_issue2()?

Reply:
It'd be great from consistency perspective.